packages/notebook-extension/src/index.ts

```diff
@@ -353,6 +356,110 @@ export const commandEditItem: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin that provides a execution indicator item to the status bar or toolbar.
+ */
+export const executionIndicator: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/notebook-extension:execution-indicator',
+  autoStart: true,
+  requires: [INotebookTracker, ILabShell, ITranslator],
+  optional: [IStatusBar, ISettingRegistry],
+  activate: (
+    app: JupyterFrontEnd,
+    notebookTracker: INotebookTracker,
+    labShell: ILabShell,
+    translator: ITranslator,
+    statusBar: IStatusBar | null,
+    settingRegistry: ISettingRegistry | null
+  ) => {
+    if (!statusBar) {
+      // Automatically disable if statusbar missing
+      return;
+    }
```

Comment:
But then we can default to the toolbar, right?

Reply:
Yes, updated in last commit

Reply:
Yes, updated in last commit